src/main/java/nextstep/fp/Lambda.java

```diff
@@ -27,30 +27,18 @@ public void run() {
     }
 
     public static int sumAll(List<Integer> numbers) {
-        int total = 0;
-        for (int number : numbers) {
-            total += number;
-        }
-        return total;
+        return sum(numbers, n -> true);
     }
 
     public static int sumAllEven(List<Integer> numbers) {
-        int total = 0;
-        for (int number : numbers) {
-            if (number % 2 == 0) {
-                total += number;
-            }
-        }
-        return total;
+        return sum(numbers, n -> n % 2 == 0);
     }
 
     public static int sumAllOverThree(List<Integer> numbers) {
-        int total = 0;
-        for (int number : numbers) {
-            if (number > 3) {
-                total += number;
-            }
-        }
-        return total;
+        return sum(numbers, n -> n > 3);
+    }
+
+    private static int sum(List<Integer> numbers, Conditional c) {
+        return numbers.stream().filter(c::test).reduce(0, Integer::sum);
```

Comment:
```suggestion
        return numbers.stream()
        .filter(c::test)
        .reduce(0, Integer::sum);
```

저는 가독성을 위해 한줄씩 하면 좋더라구요 ㅎ
디버깅시 브레이크 포인트 잡기도 편하더라구요!